### PR TITLE
Singleton OutputWriter cleanup, implement stop() on ResultTransformerOutputWriter

### PR DESCRIPTION
**Pull request by willmpls**
This PR addresses #500 to fix cleanup of singleton output writers along with implementing a stop() method on ResultTransformerOutputWriter.  See issue #500 for additional details.

### CODE REVIEW

File: jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/output/support/ResultTransformerOutputWriter.java [hunk 1]
```diff
@@ -66,5 +68,12 @@ public void doWrite(Server server, Query query, Iterable<Result> results) throws
 		return new ResultTransformerOutputWriter<>(new ResultValuesTransformer(new IdentityValueTransformer()), target);
 	}
 
+	public void stop() throws LifecycleException {
+		try {
+			target.stop();
+		} catch (LifecycleException ex) {
```
Comment (willmpls) [inline jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/output/support/ResultTransformerOutputWriter.java]: Agreed, fixed!

**Comment by gehel** [comment]
There is actually a few other OutputWriters that have the same issue. I have not gone through the whole list but at least HttpOutputWriter and WriterPoolOutputWriter have the same issue. If you have time to fix them as well, that would be great! Else I'll have a look as soon as I can.

**Comment by willmpls** [comment]
I can definitely help out with some of them.  Would you be able to generate the definitive list of ones that need fixing?  Just want to make sure I don't miss any.

**Comment by gehel** [comment]
I can go through the OutputWriter hierarchy and check, but at this point it is not much more work to fix them directly. I'll merge this PR already and I'll have a look at the whole hierarchy asap.

**Pull request merged**